src/entities/repos/submission_repo.py

```diff
@@ -151,22 +147,39 @@ async def update_task_state(
     await upsert_helper(session, task, FilingTaskProgressDAO)
 
 
-async def update_contact_info(session: AsyncSession, contact_info: ContactInfoDTO) -> ContactInfoDAO:
+async def update_contact_info(
+    session: AsyncSession, lei: str, filing_period: str, new_contact_info: ContactInfoDTO
+) -> ContactInfoDAO:
+    filing = await get_filing(session, lei=lei, filing_period=filing_period)
+    found_contact_info = await query_helper(session, ContactInfoDAO, filing=filing.id)
+    if found_contact_info:
+        contact_info = found_contact_info[0]
+        contact_info = filing.contact_info
+        contact_info.first_name = new_contact_info.first_name
+        contact_info.last_name = new_contact_info.last_name
+        contact_info.hq_address_street_1 = new_contact_info.hq_address_street_1
+        contact_info.hq_address_street_2 = new_contact_info.hq_address_street_2
+        contact_info.hq_address_city = new_contact_info.hq_address_city
+        contact_info.hq_address_state = new_contact_info.hq_address_state
+        contact_info.hq_address_zip = new_contact_info.hq_address_zip
+        contact_info.email = new_contact_info.email
+        contact_info.phone = new_contact_info.phone
+    else:
+        contact_info = ContactInfoDAO(
+            filing=filing.id,
+            first_name=new_contact_info.first_name,
+            last_name=new_contact_info.last_name,
+            hq_address_street_1=new_contact_info.hq_address_street_1,
+            hq_address_street_2=new_contact_info.hq_address_street_2,
+            hq_address_city=new_contact_info.hq_address_city,
+            hq_address_state=new_contact_info.hq_address_state,
+            hq_address_zip=new_contact_info.hq_address_zip,
+            email=new_contact_info.email,
+            phone=new_contact_info.phone,
+        )
     return await upsert_helper(session, contact_info, ContactInfoDAO)
```

Comment:
The upsert_helper function does all this.  It takes in an object like a DTO, and then copies all the data from the dto to a new instance of the passed in dao type.  So I think you just have to call it and not do anything else here.